### DISCUSSION TITLE
[BUG] 500 Error when getListingsByShop for listings with many product variations

### DISCUSSION THREAD

**Discussion by hyyyper** [etsy/open-api#892] [Q&A] [answered]
Something is causing 500 errors for this user:
Shop ID: 28086441

There are 500 errors when requesting 'active' and 'inactive' listings.
Everything works properly for 'draft', 'sold_out', 'expired'.

No other users are having these errors, I can't replicate the errors on my test shop, and this user doesn't know why their listings would be causing issues.

Has anyone encountered something like this before?

```
Endpoint: getListingsByShop
url: https://openapi.etsy.com/v3/application/shops/28086441/listings
```

```
URLSearchParams {
  'state' => 'active',
  'limit' => '100',
  'offset' => '0',
  'includes[]' => 'Inventory',
  'includes[]' => 'Images',
  'includes[]' => 'Videos',
  'includes[]' => 'Translations'
}

response: {
    status: 500,
    statusText: 'Internal Server Error',
    ...
}
```

Yet this request for sold_out listings works properly...

```
URLSearchParams {
  'state' => 'sold_out',
  'limit' => '100',
  'offset' => '0',
  'includes[]' => 'Inventory',
  'includes[]' => 'Images',
  'includes[]' => 'Videos',
  'includes[]' => 'Translations'
}

response: {
    count: 41,
    results: [...]
}
```

**Comment by emergingdzns** [comment, marked as answer]
@hyyyper I would assume that something in the shop's active or inactive listings is corrupted and the system isn't set to catch the error. I suggest the following steps (in this order):
1. Try the active/inactive calls without the includes and see if it errors still. if it doesn't, add them in one at a time until the error comes back. this would likely indicate that there's something wrong with one or more of those associations.
2. Try with a smaller limit size. Go real small and work your way up until it breaks. You may be able to spot the listing(s) that are causing the problem. if you get 500 error with a limit of 1, it could be that one first listing, so then try adjusting the offset to skip. If that resolves you know what listing(s) are causing the problem.
3. If you are able to narrow it down to a single or small handful of listings, you can probably troubleshoot them in the Etsy UI and possibly fix them.
4. If you still haven't been able to fix it, contact the developers@etsy.com team and provide any of the data you learn from the above steps. It may help your ticket be handled quicker.

**Reply by hyyyper** [reply]
@emergingdzns Thanks Andrew 🙂

Following your advice, I have identified the issue.
And, to my surprise, it doesn't seem to be any particular listing.

Here's what I did:
I tested including everything but inventory and it works fine.
I tested including only inventory and it breaks on 100 listings.
I tested including only inventory and it works at 79 listings, breaks on 80 listings.
(only inventory for the rest)
I tested offset 79, limit 1, to check that particular listing, works fine.
I tested offset 40, limit 40, to check that spread of listings, works fine.
I tested offset 20, limit 80, to check the number of listings, breaks.

This shop has tons of variants on each listing.
At 79 listings, the response payload is already around 40 MB!

There doesn't seem to be a problem with any of the listings in particular, but it breaks 100% of the time when I try to retrieve 80+ listings.
My guess would be that the Etsy service is hitting a memory limit and failing.

To solve this on my end, I'll be adjusting the limit requested for shops with tons of variants.

**Reply by emergingdzns** [reply]
That's great work @hyyyper. I recommend you send this same info over to developers@etsy.com so that they can look into the possible memory issues. I'd wager you're probably right in that the size of the inventory payload is likely overwhelming the system. Etsy should know that.

**Reply by hyyyper** [reply]
Will do 👍
